addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/api/ControlInfo.java

```diff
@@ -0,0 +1,206 @@
+    /** Degrees in Celsius. */
+    public Optional<Double> temp = Optional.empty();
+    public FanSpeed fanSpeed = FanSpeed.AUTO;
+    public FanMovement fanMovement = FanMovement.STOPPED;
+    /* Not supported by all units. Sets the target humidity for dehumidifying. */
+    public Optional<Integer> targetHumidity = Optional.empty();
+
+    private ControlInfo() {
+    }
+
+    public static ControlInfo parse(String response) {
+        logger.debug("Parsing string: \"{}\"", response);
+        ControlInfo info = new ControlInfo();
+
+        for (String keyValuePair : response.split(",")) {
+            if (keyValuePair.contains("=")) {
+                String[] keyValue = keyValuePair.split("=");
+                String key = keyValue[0];
+                String value = keyValue.length > 1 ? keyValue[1] : "";
+
+                switch (key) {
+                    case "pow":
+                        info.power = "1".equals(value);
+                        break;
+                    case "mode":
+                        parseInt(value).ifPresent((modeValue) -> info.mode = Mode.fromValue(modeValue));
+                        break;
+                    case "stemp":
+                        info.temp = parseDouble(value);
+                        break;
+                    case "f_rate":
+                        info.fanSpeed = FanSpeed.fromValue(value);
+                        break;
+                    case "f_dir":
+                        parseInt(value).ifPresent((fanValue) -> info.fanMovement = FanMovement.fromValue(fanValue));
+                        break;
+                    case "shum":
+                        info.targetHumidity = parseInt(value);
+                        break;
+                }
+            }
+        }
+
+        return info;
+    }
+
+    public WebTarget getParamString(WebTarget target) {
+        WebTarget webTarget = target.queryParam("pow", power ? 1 : 0).queryParam("mode", mode.getValue())
+                .queryParam("f_rate", fanSpeed.getValue()).queryParam("f_dir", fanMovement.getValue());
+
+        if (temp.isPresent()) {
+            webTarget = webTarget.queryParam("stemp", temp.get());
+        } else {
+            webTarget = webTarget.queryParam("stemp", "20.0");
+        }
+
+        if (targetHumidity.isPresent()) {
+            webTarget = webTarget.queryParam("shum", targetHumidity.get());
+        } else {
+            // For some reason this can be empty, but nothing else can be
+            webTarget = webTarget.queryParam("shum", "");
+        }
+
+        return webTarget;
+    }
+
+    private static Optional<Double> parseDouble(String value) {
+        try {
+            return Optional.of(Double.parseDouble(value));
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+    }
+
+    private static Optional<Integer> parseInt(String value) {
+        try {
+            return Optional.of(Integer.parseInt(value));
+        } catch (Exception e) {
```

Comment:
Prefer to log specific messages. I think you could do with `NumberException` here (or else `RuntimeException`.

Reply:
I think it should be enough to catch `NumberFormatException` as that is the exception that can be thrown by `parseInt`.